src/main/java/org/gridsuite/modification/server/modifications/NetworkStoreListener.java

```diff
@@ -275,4 +266,39 @@ private void flushEquipmentInfos() {
         equipmentInfosService.addAllEquipmentInfos(createdEquipments);
         equipmentInfosService.addAllEquipmentInfos(modifiedEquipments);
     }
+
+    private Set<AbstractBaseImpact> collectNetworkImpacts(Set<SimpleElementImpact> impacts) {
+        // keep Deletion impacts separatly
+        Set<AbstractBaseImpact> resImpacts = impacts.stream().filter(i -> i.getSimpleImpactType() == SimpleImpactType.DELETION).collect(Collectors.toSet());
+
+        // compute substations impact
+        if (impacts.stream().flatMap(i -> i.getSubstationIds().stream()).collect(Collectors.toSet()).size() >= collectionThreshold) {
```

Comment:
```suggestion
        if (impacts.stream().mapToInt(i -> i.getSubstationIds().size()).sum() >= collectionThreshold) {
```

Reply:
Multiple impacts could have the same substationIds that's why I use a Set here to remove duplicates before comparing to the `collectionThreshold`

Reply:
```suggestion
if (impacts.stream().flatMap(i -> i.getSubstationIds().stream()).distinct().count() >= collectionThreshold) {
```